src/hacbs/utils/commits-utils.ts

```diff
@@ -62,3 +71,52 @@ export const getCommitsFromPLRs = (plrList: PipelineRunKind[], limit?: number):
   );
   return limit && limit < sortedCommits.length ? sortedCommits.slice(0, limit) : sortedCommits;
 };
+
+export const getCommitByName = (pipelineruns: PipelineRunKind[], commitName: string): Commit => {
+  const plr = pipelineruns.find(
+    (pipelinerun) => pipelinerun.metadata.labels[PipelineRunLabel.COMMIT_LABEL] === commitName,
+  );
+  if (!plr) {
+    return null;
+  }
+  const commitSHA = plr.metadata.labels?.[PipelineRunLabel.COMMIT_LABEL];
+  const commitBranch =
+    plr.metadata.annotations?.[PipelineRunLabel.COMMIT_BRANCH_ANNOTATION]
+      ?.slice(1, -1)
+      .split(',') ?? [];
+  const commitUser = plr.metadata.labels?.[PipelineRunLabel.COMMIT_USER_LABEL];
+  const shaURL = plr.metadata.annotations?.[PipelineRunLabel.COMMIT_URL_ANNOTATION];
+  const shaTitle = plr.metadata.annotations?.[PipelineRunLabel.COMMIT_SHA_TITLE_ANNOTATION];
+  const creationTime = plr.metadata.creationTimestamp;
+  const application = plr.metadata.labels[PipelineRunLabel.APPLICATION];
+  const components = plr.metadata.labels[PipelineRunLabel.COMMIT_COMPONENT_LABEL]?.split(',') ?? [];
```

Comment:
A comma is an invalid character in a label value. This will not refer to multiple components.
You would need to collect unique component values from each pipelinerun